tests/utils/splitAtom.test.tsx

```diff
@@ -373,5 +373,64 @@ it('no error on wrong atom configs (fix 510)', async () => {
   fireEvent.click(evenCheckboxEl)
 
   expect(numbersEl.textContent).toBe('0')
-  expect(console.warn).toHaveBeenCalledTimes(1)
+  expect(console.warn).toHaveBeenCalledTimes(0)
+})
+
+it('variable sized splitted atom', async () => {
+  const collectionAtom = atom<number[]>([])
+  const collectionAtomsAtom = splitAtom(collectionAtom)
+
+  const derivativeAtom = atom((get) =>
+    get(collectionAtomsAtom).map(
+      (ca) => get(ca) + Math.round(Math.random() * 150)
+    )
+  )
+
+  const numberAtom = atom(1)
+
+  const generateCollection = (number: number) =>
+    Array.from({ length: Math.round(Math.random() * 30) }, (_, i) => i + number)
```

Comment:
Can we somehow make the test less complicated? Do we really need to create a random length of array?